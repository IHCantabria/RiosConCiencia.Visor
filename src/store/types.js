//DATAHUB

//Mutations
export const M_SET_PRODUCTS = "datahub/M_SET_PRODUCTS";
export const M_SET_SOURCES = "datahub/M_SET_SOURCES";
export const M_SET_PRODUCTMOMENTS = "datahub/M_SET_PRODUCTMOMENTS";
export const M_SET_PRODUCTVARIABLETYPES = "datahub/M_SET_PRODUCTVARIABLETYPES";
export const M_UPDATE_PRODUCTS_SELECTED = "datahub/M_UPDATE_PRODUCTS_SELECTED";

//Actions
export const A_FETCH_DATAHUB_PRODUCTS = "datahub/A_FETCH_DATAHUB_PRODUCTS";
export const A_FETCH_DATAHUB_SOURCES = "datahub/A_FETCH_DATAHUB_SOURCES";
export const A_FETCH_DATAHUB_PRODUCTMOMENTS =
  "datahub/M_SET_DATAHUB_PRODUCTMOMENTS";
export const A_FETCH_DATAHUB_PRODUCTVARIABLETYPES =
  "datahub/A_FETCH_DATAHUB_PRODUCTVARIABLETYPES";
export const A_SET_CLEAR_PRODUCTS = "datahub/A_SET_CLEAR_PRODUCTS";
export const A_UPDATE_PRODUCTS_SELECTED = "datahub/A_UPDATE_PRODUCTS_SELECTED";

//Getters
export const G_GET_PRODUCTS_SORTED = "datahub/G_GET_PRODUCTS_SORTED";
export const G_GET_PRODUCTS_SELECTED = "datahub/G_GET_PRODUCTS_SELECTED";
export const G_GET_PRODUCTS_COUNT = "datahub/G_GET_PRODUCTS_COUNT";
export const G_GET_PRODUCT_BY_MAPID = "datahub/G_GET_PRODUCT_BY_MAPID";
export const G_GET_SOURCE_NAMESHORT = "datahub/G_GET_SOURCE_NAMESHORT";
export const G_GET_SOURCE_TYPE = "datahub/G_GET_SOURCE_TYPE";
export const G_GET_PRODUCTMOMENTS_SORTED =
  "datahub/G_GET_PRODUCTMOMENTS_SORTED";
export const G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS =
  "datahub/G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS";

//APISENSORS

//Mutations
export const M_SET_SENSORS = "apiSensors/M_SET_SENSORS";
export const M_SET_BUOYS = "apiSensors/M_SET_BUOYS";
export const M_SET_STATIONS = "apiSensors/M_SET_STATIONS";
export const M_SET_BUOYTYPES = "apiSensors/M_SET_BUOYTYPES";
export const M_SET_STATIONTYPES = "apiSensors/M_SET_STATIONTYPES";

//Actions
export const A_FETCH_APISENSORS_BUOYS = "apiSensors/A_FETCH_APISENSORS_BUOYS";
export const A_FETCH_APISENSORS_STATIONS =
  "apiSensors/A_FETCH_APISENSORS_STATIONS";
export const A_FETCH_APISENSORS_BUOYTYPES =
  "apiSensors/A_FETCH_APISENSORS_BUOYTYPES";
export const A_FETCH_APISENSORS_STATIONTYPES =
  "apiSensors/A_FETCH_APISENSORS_STATIONTYPES";
export const A_SET_SENSORS = "apiSensors/A_SET_SENSORS";
export const A_SET_CLEAR_SENSORS = "apiSensors/A_SET_CLEAR_SENSORS";

//Getters
export const G_GET_SENSORS_SORTED = "apiSensors/G_GET_SENSORS_SORTED";
export const G_GET_SENSORS_SELECTED = "apiSensors/G_GET_SENSORS_SELECTED";
export const G_GET_SENSORS_COUNT = "apiSensors/G_GET_SENSORS_COUNT";
export const G_GET_BUOYTYPE_NAME = "apiSensors/G_GET_BUOYTYPE_NAME";

//COORDS

//Mutations
export const M_SET_SEARCH_COORDS = "coords/M_SET_SEARCH_COORDS";
export const M_SET_GO_TO_COORDS = "coords/M_SET_GO_TO_COORDS";

//Actions
export const A_SET_SEARCH_COORDS = "coords/A_SET_SEARCH_COORDS";
export const A_SET_GO_TO_COORDS = "coords/A_SET_GO_TO_COORDS";

//FILTERS

//Mutations
export const M_SET_FILTER_MOMENT = "filters/M_SET_FILTER_MOMENT";
export const M_SET_FILTER_VARIABLE = "filters/M_SET_FILTER_VARIABLE";
export const M_SET_FILTER_BUOYSTYPE = "filters/M_SET_FILTER_BUOYSTYPE";
export const M_SET_FILTER_STATIONSTYPE = "filters/M_SET_FILTER_STATIONSTYPE";
export const M_SET_FILTER_ONLY_SPATIAL_FILTERS =
  "filters/M_SET_FILTER_ONLY_SPATIAL_FILTERS";
export const M_SET_FILTER_ALLOW_PRODUCTS =
  "filters/M_SET_FILTER_ALLOW_PRODUCTS";
export const M_SET_FILTER_ALLOW_BUOYS = "filters/M_SET_FILTER_ALLOW_BUOYS";
export const M_SET_FILTER_ALLOW_STATIONS =
  "filters/M_SET_FILTER_ALLOW_STATIONS";

//Actions
export const A_SET_FILTER_MOMENT = "filters/A_SET_FILTER_MOMENT";
export const A_SET_FILTER_VARIABLE = "filters/A_SET_FILTER_VARIABLE";
export const A_SET_FILTER_BUOYSTYPE = "filters/A_SET_FILTER_BUOYSTYPE";
export const A_SET_FILTER_STATIONSTYPE = "filters/A_SET_FILTER_STATIONSTYPE";
export const A_SET_FILTER_ONLY_SPATIAL_FILTERS =
  "filters/A_SET_FILTER_ONLY_SPATIAL_FILTERS";
export const A_SET_FILTER_ALLOW_PRODUCTS = "filters/_SET_FILTER_ALLOW_PRODUCTS";
export const A_SET_FILTER_ALLOW_BUOYS = "filters/A_SET_FILTER_ALLOW_BUOYS";
export const A_SET_FILTER_ALLOW_STATIONS =
  "filters/A_SET_FILTER_ALLOW_STATIONS";

//GETTERS
export const G_GET_FILTERS_READY = "filters/G_GET_FILTERS_READY";
