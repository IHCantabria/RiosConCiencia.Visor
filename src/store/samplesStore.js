import { defineStore } from "pinia";
import {
  getYears,
  getCampaignFilter,
  getDownloadFilters,
} from "@/utils/data-operator-utils.js";
import { CAMPAIGN_VALUES } from "@/config/data-config";
import { ADMIN_ROLE_ID } from "@/config/role-config.js";
import { shallowRef } from "vue";

export const useSamplesStore = defineStore("samplesStore", {
  state: () => {
    return {
      samples: null,
      samplesPicts: null,
      yearOptions: [],
      yearFilter: {},
      campaignOptions: [],
      campaignFilter: {},
      downloadOptions: [],
      downloadFilters: [],
      user: null,
      riverSectionHistoricData: {
        active: false,
        riverSectionId: null,
      },
      riversGeoJson: shallowRef(null),
    };
  },
  getters: {
    getSamplesFiltered() {
      return this.samples
        ? this.samples.filter(
            (sample) =>
              sample.year == this.yearFilter.name &&
              sample.idCampaign == this.campaignFilter.id,
          )
        : [];
    },
    getSamplesFilteredDownloadIds() {
      if (!this.samples) {
        return [];
      } else if (
        !this.downloadFilters.length ||
        (this.downloadFilters.length == 1 && this.downloadFilters[0].id == 0)
      ) {
        return this.samples.map((sample) => sample.id);
      } else {
        return this.samples
          .filter((sample) => this.getFiltersIds.includes(sample.idFilter))
          .map((sample) => sample.id);
      }
    },
    getSamplesPictsFilteredDownloadIds() {
      if (!this.samplesPicts) {
        return [];
      } else if (
        !this.downloadFilters.length ||
        (this.downloadFilters.length == 1 && this.downloadFilters[0].id == 0)
      ) {
        return this.samplesPicts.map((sample) => sample.id);
      } else {
        return this.samplesPicts
          .filter((sample) => this.getFiltersIds.includes(sample.idFilter))
          .map((sample) => sample.id);
      }
    },
    getFiltersIds() {
      return this.downloadFilters.length
        ? this.downloadFilters.map((filter) => filter.id)
        : [];
    },
    getIsFilterAll() {
      return this.downloadFilters[0].id == 0;
    },
    getUserToken() {
      return this.user ? this.user.token : "";
    },
    getUserLogged() {
      return this.user;
    },
    getAdminLogged() {
      return this.user && this.user.roleId == ADMIN_ROLE_ID;
    },
    getSamplesLoaded() {
      return this.samples;
    },
  },
  actions: {
    setSamples(operation) {
      if (!operation) return;
      this.samples = operation;
      const years = getYears(operation);
      const filterCampaign = getCampaignFilter(
        operation,
        years[0],
        CAMPAIGN_VALUES,
      );
      this.yearOptions = years;
      this.yearFilter = years[0];
      this.campaignOptions = CAMPAIGN_VALUES;
      this.campaignFilter = filterCampaign;
      const downloadOptions = getDownloadFilters(years, CAMPAIGN_VALUES);
      this.downloadOptions = downloadOptions;
      this.downloadFilters = [downloadOptions[0]];
    },
    setPictsSamples(operation) {
      this.samplesPicts = operation;
    },
    setRiversGeoJson(operation) {
      this.riversGeoJson = operation;
    },
    setUser(operation) {
      this.user = operation;
    },
    setSampleCampaignFilter(operation) {
      this.campaignFilter = operation;
    },
    setSampleYearFilter(operation) {
      this.yearFilter = operation;
    },
    setSampleDownloadFilters(operation) {
      this.downloadFilters = operation;
    },
    async fetchSamplesAdmin(operation) {
      this.setSamplesAdmin(operation);
    },
    async fetchSamples(operation) {
      this.setSamples(operation);
    },
    async fetchPictsSamples(operation) {
      this.setPictsSamples(operation);
    },
    async fetchUser(operation) {
      this.setUser(operation);
    },
    async setSampleCampaignFilterAction(operation) {
      this.setSampleCampaignFilter(operation);
    },
    async setSampleYearFilterAction(operation) {
      this.setSampleYearFilter(operation);
    },
    async setSampleDownloadFiltersAction(operation) {
      this.setSampleDownloadFilters(operation);
    },
    setRiverSectionHistoricData(idRiverSection) {
      if (!idRiverSection) {
        this.riverSectionHistoricData.active = false;
        this.riverSectionHistoricData.riverSectionId = null;
        return;
      }
      this.riverSectionHistoricData.active = true;
      this.riverSectionHistoricData.riverSectionId = idRiverSection;
    },
  },
});
