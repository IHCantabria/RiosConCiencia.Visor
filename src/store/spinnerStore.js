import { defineStore } from "pinia";
import { spinnerConfig, spinnerEnums } from "@/config/spinner-config";

export const useSpinnerStore = defineStore("spinnerStore", {
  state: () => {
    return {
      mlActive: false,
      calls: 0,
      spinnerConfig: spinnerConfig,
      spinnerEnums: spinnerEnums,
    };
  },
  getters: {
    mlTasksConfig() {
      return this.spinnerConfig?.tasksConfig || null;
    },
  },
  actions: {
    show() {
      this.calls++;
      this.mlActive = true;
    },
    hide() {
      this.calls--;
      if (this.calls == 0) {
        this.mlActive = false;
      }
      if (this.calls == 0) this._clearTasks();
    },
    configureModeSpinner() {
      this.spinnerConfig.mode = this.spinnerEnums.ModesSpinner.progress;
    },
    addTask(name, text) {
      this.spinnerConfig.tasksConfig.tasks.push({
        id: this.spinnerConfig.tasksConfig.tasks.length + 1,
        name,
        text,
        status: this.spinnerEnums.statusTask.pending,
      });
    },
    updateTask(name, status = null, text = null) {
      const indexTask = this.spinnerConfig.tasksConfig.tasks.findIndex(
        (task) => task.name === name,
      );
      if (status) {
        this.spinnerConfig.tasksConfig.tasks[indexTask].status = status;
      }
      if (text) {
        this.spinnerConfig.tasksConfig.tasks[indexTask].text = text;
      }
    },
    _clearTasks() {
      this.spinnerConfig.tasksConfig.tasks = [];
    },
  },
});
