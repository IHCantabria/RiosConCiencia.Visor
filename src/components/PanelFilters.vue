<template>
  <div v-if="moments" class="app-filters">
    <template v-if="moments">
      <filters-advanced
        class="filters-advanced"
        data-cy="filtersAdvancedToggle"
        @only-spatial-filters-actived="toggleSpatialFilters"
        @products-actived="toggleProducts"
        @buoys-actived="toggleBuoys"
        @stations-actived="toggleStations"
      ></filters-advanced>
      <panel-about class="panel-about"></panel-about>
      <div
        :class="[
          computedShowTypeFilters
            ? 'type-filters__expanded'
            : 'type-filters__reduced'
        ]"
      >
        <div class="time-selectors">
          <slider
            class="time-selectors__selector"
            :range-data="moments"
            theme="light"
            @change="setMomentFilter"
          ></slider>
        </div>
        <div class="variable-selectors">
          <vs-tabs v-if="momentFilter" ref="filtersTabs" position="left">
            <vs-tab label="Products">
              <variable-selector
                v-if="productTypes"
                class="variable-selectors__selector"
                :items="productTypes"
                :last-active-item="variableFilter"
                @change="setVariableFilter"
              ></variable-selector>
            </vs-tab>
            <vs-tab label="Sensors">
              <div class="type-selectors">
                <type-selector
                  v-if="buoyTypes"
                  class="type-selectors__selector"
                  label="Buoy Type Selector"
                  :width-select="filtersWidth"
                  :items="buoyTypes"
                  :last-active-item="buoyTypeFilter"
                  @change="setBuoyTypeFilter"
                ></type-selector>
                <type-selector
                  v-if="stationTypes"
                  class="type-selectors__selector"
                  label="Station Type Selector"
                  :width-select="filtersWidth"
                  :items="stationTypes"
                  :last-active-item="stationTypeFilter"
                  @change="setStationTypeFilter"
                ></type-selector>
              </div>
            </vs-tab>
          </vs-tabs>
        </div>
      </div>
    </template>
  </div>
</template>
<script>
import { mapActions, mapState, mapGetters } from "vuex";
import * as types from "@/store/types";
import rangeSlider from "@/Lib/components/RangeSlider";
import radioButtonList from "@/Lib/components/RadioButtonList";
import BaseSelectorType from "@/components/ui/BaseSelectorType";
import BaseToggleFilters from "@/components/ui/BaseToggleFilters";
import AppAbout from "@/components/layout/AppAbout.vue";
export default {
  components: {
    slider: rangeSlider,
    "variable-selector": radioButtonList,
    "type-selector": BaseSelectorType,
    "filters-advanced": BaseToggleFilters,
    "panel-about": AppAbout
  },
  props: {
    filtersWidth: {
      type: String,
      required: false,
      default: () => {
        return "200px";
      }
    }
  },
  data() {
    return {
      showTypeFilters: false
    };
  },
  computed: {
    ...mapState({
      stationTypeFilter: state => state.filters.stationTypeFilter,
      buoyTypeFilter: state => state.filters.buoyTypeFilter,
      variableFilter: state => state.filters.variableFilter,
      momentFilter: state => state.filters.momentFilter,
      buoyTypes: state => state.apiSensors.buoyTypes,
      stationTypes: state => state.apiSensors.stationTypes
    }),
    ...mapGetters({
      moments: [types.G_GET_PRODUCTMOMENTS_SORTED],
      productTypes: [types.G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS]
    }),
    computedShowTypeFilters() {
      return this.showTypeFilters ? false : true;
    }
  },
  watch: {
    momentFilter: {
      handler() {
        this.resetSelections();
        this.stationTypes
          ? this.setStationTypeFilter(this.stationTypes[0])
          : "";
        this.buoyTypes ? this.setBuoyTypeFilter(this.buoyTypes[0]) : "";
      },
      deep: true
    },
    stationTypes: {
      handler() {
        this.setStationTypeFilter(this.stationTypes[0]);
      },
      deep: true
    },
    buoyTypes: {
      handler() {
        this.setBuoyTypeFilter(this.buoyTypes[0]);
      },
      deep: true
    }
  },
  methods: {
    ...mapActions({
      setProductsMomentFilter: types.A_SET_FILTER_MOMENT,
      setProductsVariableFilter: types.A_SET_FILTER_VARIABLE,
      setSensorsBuoyTypeFilter: types.A_SET_FILTER_BUOYSTYPE,
      setSensorsStationTypeFilter: types.A_SET_FILTER_STATIONSTYPE,
      setOnlySpatialFilters: types.A_SET_FILTER_ONLY_SPATIAL_FILTERS,
      setAllowSearchProducts: types.A_SET_FILTER_ALLOW_PRODUCTS,
      setAllowSearchBuoys: types.A_SET_FILTER_ALLOW_BUOYS,
      setAllowSearchStations: types.A_SET_FILTER_ALLOW_STATIONS
    }),
    setMomentFilter(moment) {
      this.setProductsMomentFilter(moment);
    },
    setVariableFilter(variable) {
      this.setProductsVariableFilter(variable);
    },
    setBuoyTypeFilter(type) {
      this.setSensorsBuoyTypeFilter(type);
    },
    setStationTypeFilter(type) {
      this.setSensorsStationTypeFilter(type);
    },
    toggleSpatialFilters(isActive) {
      this.showTypeFilters = isActive;
      this.setOnlySpatialFilters(isActive);
    },
    toggleProducts(isActive) {
      this.setAllowSearchProducts(isActive);
    },
    toggleBuoys(isActive) {
      this.setAllowSearchBuoys(isActive);
    },
    toggleStations(isActive) {
      this.setAllowSearchStations(isActive);
    },
    resetSelections() {
      this.$refs.filtersTabs ? this.$refs.filtersTabs.activeChild(0) : ""; //select Products Tab
      this.setProductsVariableFilter(null);
      this.setBuoyTypeFilter(null);
      this.setStationTypeFilter(null);
    }
  }
};
</script>
<style scoped lang="scss">
.app-filters {
  display: flex;
  flex-direction: column;
}
.type-filters {
  &__expanded {
    max-height: 25vh;
    transition: all 500ms ease-in 0s;
  }
  &__reduced {
    max-height: 0;
    transition: all 500ms ease-in 0s;
  }
}

.time-selectors {
  width: 120%;
  transform: scale(0.8);
  margin-top: 1%;
  margin-bottom: 5%;

  &__selector {
    width: 115%;
    transform: translateX(-15%);
  }
}

.variable-selectors {
  border-top: 1.5px solid $color-light-grey;
  width: 100%;
  padding: 1%;
  margin: 1% 0%;
  display: flex;

  &__selector {
    width: 120%;
    transform-origin: left;
    transform: scale(0.8);
  }
}

.type-selectors {
  width: 100%;
  display: flex;
  flex-flow: row wrap;

  &__selector {
    margin: 1vw;
  }
}
.filters-advanced {
  top: 0;
  left: 0;
  z-index: 1;
  position: absolute;
}
.panel-about {
  top: 0;
  right: 0;
  z-index: 1;
  position: absolute;
}
</style>
