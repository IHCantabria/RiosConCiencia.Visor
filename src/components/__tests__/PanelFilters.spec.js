import { shallowMount, createLocalVue } from "@vue/test-utils";
import PanelFilters from "../PanelFilters.vue";
import merge from "lodash.merge";
import Vuex from "vuex";
import Vuesax from "vuesax";
import * as types from "@/store/types";

const localVue = createLocalVue();
localVue.use(Vuesax);
localVue.use(Vuex);

function createWrapper(overrides) {
  const defaultMountingOptions = {
    propsData: {},
    localVue
  };
  return shallowMount(PanelFilters, merge(defaultMountingOptions, overrides));
}

describe("panelFilters.vue", () => {
  let store;
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        apiSensors: {
          state: {
            buoyTypes: null,
            stationTypes: null
          },
          getters: {
            [types.G_GET_SENSORS_SORTED]: () => []
          }
        },
        datahub: {
          getters: {
            [types.G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS]: () => []
          }
        },
        filters: {
          state: {
            variableFilter: null,
            momentFilter: null,
            stationTypeFilter: null,
            buoyTypeFilter: null
          },
          actions: {
            [types.A_SET_FILTER_MOMENT]: jest.fn(),
            [types.A_SET_FILTER_VARIABLE]: jest.fn(),
            [types.A_SET_FILTER_BUOYSTYPE]: jest.fn(),
            [types.A_SET_FILTER_STATIONSTYPE]: jest.fn()
          }
        }
      }
    });
  });
  //UNIT TESTS
  it("setProductsMomentFilter is called when setMomentFilter is called", () => {
    expect.assertions(1);
    const setProductsMomentFilter = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_MOMENT]: setProductsMomentFilter
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.setMomentFilter("t");
    expect(setProductsMomentFilter).toHaveBeenCalled();
  });
  it("setProductsVariableFilter is called when setVariableFilter is called", () => {
    expect.assertions(1);
    const setProductsVariableFilter = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_VARIABLE]: setProductsVariableFilter
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.setVariableFilter();
    expect(setProductsVariableFilter).toHaveBeenCalled();
  });
  it("setSensorsBuoyTypeFilter is called when setBuoyTypeFilter is called", () => {
    expect.assertions(1);
    const setSensorsBuoyTypeFilter = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_BUOYSTYPE]: setSensorsBuoyTypeFilter
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.setBuoyTypeFilter();
    expect(setSensorsBuoyTypeFilter).toHaveBeenCalled();
  });
  it("setSensorsStationTypeFilter is called when setStationTypeFilter is called", () => {
    expect.assertions(1);
    const setSensorsStationTypeFilter = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_STATIONSTYPE]: setSensorsStationTypeFilter
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.setStationTypeFilter();
    expect(setSensorsStationTypeFilter).toHaveBeenCalled();
  });
  it("setAllowFilters is called when toggleFilters is called", () => {
    expect.assertions(1);
    const setOnlySpatialFilters = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_ONLY_SPATIAL_FILTERS]: setOnlySpatialFilters
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.toggleSpatialFilters();
    expect(setOnlySpatialFilters).toHaveBeenCalled();
  });
  it("setAllowSearchProducts is called when toggleProducts is called", () => {
    expect.assertions(1);
    const setAllowSearchProducts = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_ALLOW_PRODUCTS]: setAllowSearchProducts
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.toggleProducts();
    expect(setAllowSearchProducts).toHaveBeenCalled();
  });
  it("setAllowSearchBuoys is called when toggleBuoys is called", () => {
    expect.assertions(1);
    const setAllowSearchBuoys = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_ALLOW_BUOYS]: setAllowSearchBuoys
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.toggleBuoys();
    expect(setAllowSearchBuoys).toHaveBeenCalled();
  });
  it("setAllowSearchStations is called when toggleStations is called", () => {
    expect.assertions(1);
    const setAllowSearchStations = jest.fn();
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      actions: {
        [types.A_SET_FILTER_ALLOW_STATIONS]: setAllowSearchStations
      }
    });
    const wrapper = createWrapper({
      store
    });
    wrapper.vm.toggleStations();
    expect(setAllowSearchStations).toHaveBeenCalled();
  });
  //SNAPSHOT TESTS
  it("renders correctly by default", () => {
    expect.assertions(1);
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      }
    });
    const wrapper = createWrapper({
      store
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders correctly with moments", () => {
    expect.assertions(1);
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => [
          {
            id: 1,
            label: "Past",
            icon: "history",
            name: "Past",
            color: "#dce8f4"
          },
          {
            id: 2,
            label: "Present",
            icon: "play_arrow",
            name: "Present",
            color: "#98c5d1"
          },
          {
            id: 3,
            label: "Short-term Forecast",
            icon: "redo",
            name: "Short term future",
            color: "#00a09e"
          },
          {
            id: 4,
            label: "Climate Change",
            icon: "last_page",
            name: "Long term future",
            color: "#19535F"
          }
        ]
      }
    });
    const wrapper = createWrapper({
      store
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly with momentFilter", () => {
    expect.assertions(1);
    store.hotUpdate({
      getters: {
        [types.G_GET_PRODUCTMOMENTS_SORTED]: () => null
      },
      state: {
        momentFilter: {}
      }
    });
    const wrapper = createWrapper({
      store
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
