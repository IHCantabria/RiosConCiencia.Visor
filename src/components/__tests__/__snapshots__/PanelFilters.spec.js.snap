// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panelFilters.vue renders correctly by default 1`] = `<!---->`;

exports[`panelFilters.vue renders correctly with momentFilter 1`] = `<!---->`;

exports[`panelFilters.vue renders correctly with moments 1`] = `
<div
  class="app-filters"
>
  <filters-advanced-stub
    class="filters-advanced"
    data-cy="filtersAdvancedToggle"
  />
   
  <panel-about-stub
    class="panel-about"
  />
   
  <div
    class="type-filters__expanded"
  >
    <div
      class="time-selectors"
    >
      <slider-stub
        class="time-selectors__selector"
        id="selector"
        rangedata="[object Object],[object Object],[object Object],[object Object]"
        theme="light"
      />
    </div>
     
    <div
      class="variable-selectors"
    >
      <!---->
    </div>
  </div>
</div>
`;
